### DISCUSSION TITLE
Trino Hive Connector specifying compression

### DISCUSSION THREAD

**Discussion by omalley** [trinodb/trino#15037] [General]
Is it possible with Trino and the Hive Connector to specify the compression used to write data in
a create table?

**Comment by hashhar** [comment]
Not as part of the `CREATE TABLE` itself but you can set the `compression_codec` catalog session property. So something like:

```sql
SET SESSION my_hive_catalog.compression_codec = 'ZSTD';
CREATE TABLE my_hive_catalog.default.my_table AS SELECT * FROM tpch.tiny.orders;
```

Note that the reason why compression config is not part of the table definition is because a table is allowed to contain whatever mix of compressions across files - even in same partition or across partitions.

This does mean that everytime you INSERT to table and you want to use same compression you need to make sure to set the compression codec in the session.
